### PR TITLE
feat: adds a fetch for hard coded spot price of future token

### PR DESCRIPTION
**Pull request by R-K-H**
moves fetch for token into a hook, adds rounding and capabilities to add different tokens for middle bar props for layout.

### CODE REVIEW

File: hooks/useFetchSpotPrice.ts [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  programKey: string | null;
+};
+
+export function useFetchSpotPrice(props: FetchSpotPriceProps) {
+  const { programKey } = props;
+  let inputMint = 'METADDFL6wWMWEoKTFJwcThTbUmtarRJZjRpzUvkxhr';
+  let tokenBase = 100_000_000;
+  let decimals = 3;
+  let tokenName = 'META';
+  if (programKey
```
Comment (swaggymarie) [inline hooks/useFetchSpotPrice.ts:12]: good enough for now but a bit nasty lol

File: app/program/page.tsx [hunk 2]
```diff
@@ -9,7 +9,7 @@ export default function DaoPage() {
   const params = useSearchParams();
   const programKey = params.get('programKey');
   return (
-    <Layout>
+    <Layout programKey={programKey}>
```
Comment (swaggymarie) [inline app/program/page.tsx]: I don't think it's super clean to do that way. I would create a TokenPrice component and useSearchParams inside this one, so nothing is passed to the layout comp

File: hooks/useFetchSpotPrice.ts (hunk 1 above)
Comment (R-K-H) [reply hooks/useFetchSpotPrice.ts:12]: SOooo nasty.

File: app/program/page.tsx (hunk 2 above)
Comment (R-K-H) [reply app/program/page.tsx]: The issue is where the component exists. In layout.. Ahh actually I see what you're saying there, the only thing is we don't need search params for ALL layouts... And then I'd have to pass that through... Dunno I feel like it's sloppy to get it done, vs break this dead cat open more...

Comment (swaggymarie) [reply app/program/page.tsx]: if you do params.get('programKey') and it's not in the path, then it would return undefined and you would not show the price

Comment (R-K-H) [reply app/program/page.tsx]: Ahhh gotcha. I do have a check for that, where it just defaults to META so it does work. But I take your point.

Comment (R-K-H) [reply app/program/page.tsx]: Fixed, ready for review.

**Review by swaggymarie — APPROVED** [review]

**Pull request merged**